sonar-delphi-plugin/src/main/java/au/com/integradev/delphi/DelphiRulesDefinition.java

```diff
@@ -53,9 +58,19 @@ public void define(Context context) {
 
     RuleTemplateAnnotationReader ruleTemplateAnnotationReader = new RuleTemplateAnnotationReader();
 
-    ruleMetadataLoader.addRulesByAnnotatedClass(repository, CheckList.getChecks());
-    ruleTemplateAnnotationReader.updateRulesByAnnotatedClass(repository, CheckList.getChecks());
+    List<Class<?>> checks =
+        CheckList.getChecks().stream()
+            .filter(this::isCheckSupportedOnPlatform)
+            .collect(Collectors.toList());
+
+    ruleMetadataLoader.addRulesByAnnotatedClass(repository, checks);
+    ruleTemplateAnnotationReader.updateRulesByAnnotatedClass(repository, checks);
 
     repository.done();
   }
+
+  private boolean isCheckSupportedOnPlatform(Class<?> check) {
+    return runtime.getProduct() == SonarProduct.SONARQUBE
+        || AnnotationUtils.getAnnotation(check, SonarLintUnsupported.class) == null;
```

Comment:
This is more explicit and targeted to the sonarlint context.
```suggestion
    return !(runtime.getProduct() == SonarProduct.SONARLINT
        && AnnotationUtils.getAnnotation(check, SonarLintUnsupported.class) != null);
```

Reply:
That seems much more confusing to me - instead of having no negations, it has two at different levels.

I copied the approach in DelphiChecksExecutor - see:

https://github.com/integrated-application-development/sonar-delphi/blob/e4d61b6479e86e177b4ac149e6305e9d7292e844/delphi-frontend/src/main/java/au/com/integradev/delphi/executor/DelphiChecksExecutor.java#L102-L105

Reply:
I suppose the way you suggest is more resilient to future additions to the `SonarProduct` enum. Should I change DelphiChecksExecutor as well?

Reply:
Huh. I'm surprised that I wrote it that way in the checks executor...
I think it's fine, let's just leave it as-is for now.
A new `SonarProduct` with special rules would be a big change regardless and prompt us to review `SonarProduct` usages anyway.